File: nogithubbay003_ui/src/services/AdvertisementService.js
import http from "../http-common"; 

class AdvertisementService {
  getAllAdvertisements(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/advertisement/advertisements`, searchDTO);
  }

  get(advertisementId) {
    return this.getRequest(`/advertisement/${advertisementId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/advertisement?field=${matchData}`, null);
  }

  addAdvertisement(data) {
    return http.post("/advertisement/addAdvertisement", data);
  }

  update(data) {
  	return http.post("/advertisement/updateAdvertisement", data);
  }
  
  uploadImage(data,advertisementId) {
  	return http.postForm("/advertisement/uploadImage/"+advertisementId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new AdvertisementService();

File: src/main/java/com/nogithubbay003/service/AdvertisementService.java
package com.nogithubbay003.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.nogithubbay003.domain.Advertisement;
import com.nogithubbay003.dto.AdvertisementDTO;
import com.nogithubbay003.dto.AdvertisementSearchDTO;
import com.nogithubbay003.dto.AdvertisementPageDTO;
import com.nogithubbay003.dto.AdvertisementConvertCriteriaDTO;
import com.nogithubbay003.service.GenericService;
import com.nogithubbay003.dto.common.RequestDTO;
import com.nogithubbay003.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface AdvertisementService extends GenericService<Advertisement, Integer> {

	List<Advertisement> findAll();

	ResultDTO addAdvertisement(AdvertisementDTO advertisementDTO, RequestDTO requestDTO);

	ResultDTO updateAdvertisement(AdvertisementDTO advertisementDTO, RequestDTO requestDTO);

    Page<Advertisement> getAllAdvertisements(Pageable pageable);

    Page<Advertisement> getAllAdvertisements(Specification<Advertisement> spec, Pageable pageable);

	ResponseEntity<AdvertisementPageDTO> getAdvertisements(AdvertisementSearchDTO advertisementSearchDTO);
	
	List<AdvertisementDTO> convertAdvertisementsToAdvertisementDTOs(List<Advertisement> advertisements, AdvertisementConvertCriteriaDTO convertCriteria);

	AdvertisementDTO getAdvertisementDTOById(Integer advertisementId);







}





